src/compiler/crystal/util.cr

```diff
@@ -17,28 +13,28 @@ module Crystal
     filename
   end
 
-  def self.error(msg, color, exit_code = 1, stderr = STDERR)
-    stderr.print "Error: ".colorize.toggle(color).red.bold
-    stderr.puts msg.colorize.toggle(color).bright
+  def self.error(msg, exit_code = 1, stderr = STDERR)
+    stderr.print "Error: ".colorize.red.bold
+    stderr.puts msg.colorize.bright
     exit(exit_code) if exit_code
   end
 
   def self.tempfile(basename)
     CacheDir.instance.join("crystal-run-#{basename}.tmp")
   end
 
-  def self.with_line_numbers(source : String, highlight_line_number = nil, color = false)
-    source.lines.map_with_index do |line, i|
-      str = "#{"%4d" % (i + 1)}. #{line.to_s.chomp}"
-      target = i + 1 == highlight_line_number
-      if target
-        if color
-          str = ">".colorize.green.bold.to_s + str[1..-1].colorize.bold.to_s
+  def self.with_line_numbers(source : String, highlight_line_number = nil)
+    String.build do |io|
+      i = 0
+      source.each_line do |line|
```

Comment:
So `source.lines.each_with_index(1)`? That'd be slower since it would first iterate through `source`'s lines and put it in array and then iterate through it again with an index while `source.each_line do |line|` only iterates through the lines.

Reply:
Oh, never mind. I'll add a proposal in a separate PR.